core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java

```diff
@@ -217,6 +220,14 @@ public BitSet prunePages(RawBlockletColumnChunks rawBlockletColumnChunks)
           bitSet.set(i);
         }
       }
```

Comment:
when `carbon.push.rowfilters.for.vector` = `true`,  This method won't be called. 
Need to handle same in ```
applyFilter(RawBlockletColumnChunks rawBlockletColumnChunks,
      boolean useBitsetPipeLine)
```

Reply:
I know that, as written in the description of this PR. However, applyFilter and prunePages is quite similar, I am wondering if anyone can refactor this part first.

Reply:
I saw that description now. 

**Whether you do refactor or not need to handle in the apply filter now.** 

If possible you can refactor also. If not just handle your scenario , later we can refactor

Reply:
ok I will handle in `applyFilter`

Reply:
I re-think about this.Different to minmax, page bloom costs more(storage, decode) when query. Row level filter may not benefit from bloom.

If we use **one filter column** and get multiple columns, once bloom says that page does not need to scan, nothing need to do for all columns of this page in direct fill case. If bloom can skip more pages, the IO benefit for skipped pages is `# of pages * project columns`. 

As for row filter, for same query, the benefit is shrank to `# of pages * 1 (the filter column)`. 

For row level filter:
```
1.original: 

decode page -> for-loop checking each value of this column

2.with page bloom: 

read bloom chunk -> decode bloom bitmap -> check bloom

if check result is false -> skip
if check result is true ->  decode page -> for-loop checking each value of this column
```

Reply:
yeah. make sense. Let me check again

Reply:
But it will be useful If data doesn't exist in page. we can skip the page, avoiding checking every row of that page.
However If the data exist in page. It is an extra operation.
This problem exist in prunePages() also. we decode bloom chunk , even though data exist  in page.

Reply:
That sounds like some gambling. Bloom check v.s. Direct scan. Skip pages v.s. Extra Operation.
Need some more test.